contracts/testing/test_request.go

```diff
@@ -1,10 +1,22 @@
 package testing
 
+import (
+	"context"
+	"io"
+)
+
 type TestRequest interface {
 	WithHeaders(headers map[string]string) TestRequest
 	WithHeader(key, value string) TestRequest
 	WithoutHeader(key string) TestRequest
 	WithCookies(cookies map[string]string) TestRequest
 	WithCookie(key, value string) TestRequest
+	WithContext(ctx context.Context) TestRequest
 	Get(uri string) (TestResponse, error)
+	Post(uri string, body io.Reader) (TestResponse, error)
```

Comment:
`body io.Reader` is a bit difficult to build. Can we provide a simple way to do so?

<img width="706" alt="image" src="https://github.com/user-attachments/assets/40939f9f-4733-4161-8fff-86afab6734c3">

Reply:
Sure, we can provide a interface to write body.

Reply:
We can optimize it in another PR if you want.